### PR TITLE
Home page now shows different backgrounds depending on time of day

### LINKED ISSUE #20: FEATURE: Changing Home Page
## User Story
After this change, I will:
- be able to have the commons background change depending on when I log in.

## Discussion
To make it more exciting, I think we can add a few more background assets and change what is displayed based on local PST time (aka where the farm is). That way, the commons page isn't just a single static image, it is more interactive and gives a sense of reality to it. There are tons of copyright-free images from [Unsplash](https://unsplash.com/) that we can use for this.

### PR DESCRIPTION
**Pull request by dscpsyl**
## Overview
In this PR, the homepage will show a different background depending on the time of day in local time. This helps with user experience and improves the overall novelness of the game. A nice-to-have feature. All images are from [Unsplash](https://unsplash.com/) and are thus free for use. 

## Screenshots (Optional)
Screenshot 2023-08-15 at 11 44 59 PM

## Storybook links

* https://ucsb-cs156-m23.github.io/proj-happycows-m23-10am-4/prs/58/storybook/?path=/story/pages-homepage--morning
* https://ucsb-cs156-m23.github.io/proj-happycows-m23-10am-4/prs/58/storybook/?path=/story/pages-homepage--day
* https://ucsb-cs156-m23.github.io/proj-happycows-m23-10am-4/prs/58/storybook/?path=/story/pages-homepage--evening
* https://ucsb-cs156-m23.github.io/proj-happycows-m23-10am-4/prs/58/storybook/?path=/story/pages-homepage--night

## Feedback Request (Optional)
If you have any ideas of what better images to use, please feel free to share. Please also reference the current issue for the timing of background changes and let me know if you disagree with them. 

## Tests
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #20

### CODE REVIEW

**Comment by iain801** [comment]
Looks like many mutants are surviving in the .js files.

**Comment by dscpsyl** [comment]
@iain801 Should be fix now

**Review by iain801 — DISMISSED** [review]
Looks good to me, works on dokku-qa!

**Review by yuanchenmeng — DISMISSED** [review]
LGTM!

File: frontend/src/main/pages/HomePage.js [hunk 1]
```diff
@@ -6,8 +6,8 @@ import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
 import CommonsList from "main/components/Commons/CommonsList";
 import { useBackend, useBackendMutation } from "main/utils/useBackend";
 import { useCurrentUser } from "main/utils/currentUser";
-import Background from './../../assets/HomePageBackground.jpg';
 import { commonsNotJoined } from "main/utils/commonsUtils";
+import getBackgroundImage from "main/components/Utils/HomePageBackground";
 
 export default function HomePage() {
```
Comment (pconrad) [inline frontend/src/main/pages/HomePage.js]:
```suggestion
export default function HomePage({hour=null}) {
```

This suggestion goes along with the one below.

File: frontend/src/main/pages/HomePage.js [hunk 2]
```diff
@@ -53,9 +53,13 @@ export default function HomePage() {
   const visitButtonClick = (id) => { navigate("/play/" + id) };
 
   //create a list of commons that the user hasn't joined for use in the "Join a New Commons" list.
- 
   const commonsNotJoinedList = commonsNotJoined(commons, commonsJoined);
+
+  // Get the current time and set the background image accordingly
+  const time = new Date().getHours();
```
Comment (pconrad) [inline frontend/src/main/pages/HomePage.js]: This suggestion goes along with the one above, to have a parameter hour that, by default, is null, but can be overridden (this is for use with storybook).   Unless I'm mistaken, these two suggestions won't change any existing functionality but will allow you to add some stories to show what the home page will look like at different times of day.

```suggestion
  const time = ( hour===null)  ? new Date().getHours() : hour;
```

**Review by pconrad — CHANGES_REQUESTED** [review]
I've suggested a couple of changes that will allow you to show off this feature in the storybook.  That's helpful, because otherwise, there's very little way to show it off.

See the changes that I suggested in the HomePage.js component first. Then, we can add something like this in:



```js

const Template = (args) => <HomePage  {...args} />;

export const Morning = Template.bind({});
Morning.args = {
    hour: 8
};

export const Day = Template.bind({});
Day.args = {
    hour: 10
};

// etc for Evening and Night

```

**Comment by dscpsyl** [comment]
@pconrad Thank you for the suggestions! Storybook changes have been implemented and are now working.

**Review by iain801 — APPROVED** [review]
I'm assuming the @phtcon's fixes work, lgtm!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**